swaybar/main.c

```diff
@@ -1034,6 +1067,44 @@ int i3json_handle_fd(int fd) {
 	return i3json_parse();
 }
 
+bool handle_ipc_event() {
+	struct ipc_response *resp = ipc_recv_response(ipc_event_socketfd);
+	switch (resp->type) {
+	case IPC_EVENT_WORKSPACE:
+		ipc_update_workspaces();
+		break;
+	case IPC_EVENT_MODE: ;
```

Comment:
You can move the declaration, or you could add curly braces around the body of this case block.